src/Infrastructure/Processor/Process/MediaTranslationShopware6ExportProcess.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ergonode\ExporterShopware6\Infrastructure\Processor\Process;
+
+use Ergonode\Channel\Domain\Entity\Export;
+use Ergonode\Channel\Domain\Repository\ExportRepositoryInterface;
+use Ergonode\ExporterShopware6\Domain\Entity\Shopware6Channel;
+use Ergonode\ExporterShopware6\Domain\Repository\LanguageRepositoryInterface;
+use Ergonode\ExporterShopware6\Domain\Repository\ProductRepositoryInterface;
+use Ergonode\ExporterShopware6\Infrastructure\Connector\Action\Media\GetMediaTranslations;
+use Ergonode\ExporterShopware6\Infrastructure\Connector\Action\Media\PatchMediaAction;
+use Ergonode\ExporterShopware6\Infrastructure\Connector\Shopware6Connector;
+use Ergonode\ExporterShopware6\Infrastructure\Connector\Shopware6QueryBuilder;
+use Ergonode\ExporterShopware6\Infrastructure\Model\Media\Shopware6MediaTranslation;
+use Ergonode\ExporterShopware6\Infrastructure\Model\Shopware6Media;
+use Ergonode\Multimedia\Domain\Entity\AbstractMultimedia;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Log\LoggerInterface;
+use Throwable;
+use Webmozart\Assert\Assert;
+
+class MediaTranslationShopware6ExportProcess
+{
+    private Shopware6Connector $connector;
+
+    protected ProductRepositoryInterface $shopwareProductRepository;
+
+    private LanguageRepositoryInterface $languageRepository;
+
+    private ExportRepositoryInterface $exportRepository;
+
+    private LoggerInterface $logger;
+
+    public function __construct(
+        Shopware6Connector $connector,
+        LanguageRepositoryInterface $languageRepository,
+        ExportRepositoryInterface $exportRepository,
+        ProductRepositoryInterface $shopwareProductRepository,
+        LoggerInterface $logger
+    ) {
+        $this->connector = $connector;
+        $this->languageRepository = $languageRepository;
+        $this->exportRepository = $exportRepository;
+        $this->shopwareProductRepository = $shopwareProductRepository;
+        $this->logger = $logger;
+    }
+
+    public function process(
+        Export $export,
+        Shopware6Channel $channel,
+        AbstractMultimedia $multimedia,
+        string $shopwareId
+    ): void {
+        try {
+            $this->ensureMediaTranslations($channel, $multimedia, $shopwareId);
+        } catch (Throwable $exception) {
+            $this->exportRepository->addError($export->getId(), $exception->getMessage(), []);
+        }
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    private function ensureMediaTranslations(
+        Shopware6Channel $channel,
+        AbstractMultimedia $multimedia,
+        string $shopwareId
+    ): void {
+        //apply translations from shopware
+        $shopware6Media = new Shopware6Media($shopwareId, null);
```

Comment:
I think we should check in the repository for shopware6 media whether this shopwareid is stored in our database